### PR TITLE
Consolidate go-billy and create new osfs2 (from `fluxcd/pkg/git/gogit/fs`)

### PR DESCRIPTION
**Pull request by pjbgf**
The aim of this PR is to consolidate the go-billy implementation for Flux.

A new FS `osfs2` implementation was introduced, which provides OS filesystem support, without an user-space chroot.
The original (`fluxcd/pkg/git/gogit/fs`) code required some minor changes in order to improve its Windows support, and to align with `osfs` the multi-platform support for operations such as lock/unlock.

### CODE REVIEW

**Comment by hiddeco** [comment]
Think `fluxcd/pkg/fs` refers to `fluxcd/pkg/git/gogit/fs`?

File: README.md [hunk 1]
```diff
@@ -8,14 +8,14 @@ Billy was born as part of [go-git/go-git](https://github.com/go-git/go-git) proj
 ## Installation
 
 ```go
-import "github.com/go-git/go-billy/v5" // with go modules enabled (GO111MODULE=on or outside GOPATH)
-import "github.com/go-git/go-billy" // with go modules disabled
+import "github.com/fluxcd/go-billy/v5" // with go modules enabled (GO111MODULE=on or outside GOPATH)
```
Comment (hiddeco) [inline README.md]: What is the reasoning behind changing the module name? Given our goals are not to create a hard fork, I would say this is a conflicting choice (and signals something different). If the argument is versioning, I think Go Modules `replace` works just fine with tags (`replace github.com/go-git/go-billy/v5 => github.com/fluxcd/go-billy/v5 <tag on fork>`).

Comment (pjbgf) [reply README.md]: It was mostly to align with the go-git approach. I have now reverted this change.

File: osfs2/os.go [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+	if !filepath.IsAbs(link) {
+		link = filepath.Clean(filepath.Join(fs.workingDir, link))
+	}
+	if ok, err := fs.insideWorkingDirEval(link); !ok {
+		return "", err
+	}
+	return os.Readlink(link)
+}
+
+// Chroot returns a new OS filesystem, with working directory set to path.
```
Comment (aryan9600) [inline osfs2/os.go]:
```suggestion
// Chroot returns a new OS filesystem, with the working dir set to the result of joining the provided path with the underlying working dir.
```

File: README.md (hunk 1 above)
Comment (hiddeco) [reply README.md]: Yes, I was too late with jumping in on that change :grimacing:

**Comment by pjbgf** [comment]
Any more changes required for this PR?

File: go.mod [hunk 3]
```diff
@@ -1,10 +1,18 @@
...
+require (
+	github.com/google/go-cmp v0.5.9 // indirect
+	github.com/kr/pretty v0.2.1 // indirect
+	github.com/kr/text v0.2.0 // indirect
+	golang.org/x/net v0.4.0 // indirect
+	golang.org/x/text v0.5.0 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+go 1.19
```
Comment (hiddeco) [inline go.mod]: Would put this at top.

**Comment by pjbgf** [comment]
Changes made and bumped dependencies. PTAL

**Review by hiddeco — APPROVED** [review]
Looks good to me, sorry for letting this slip through the cracks :bow:

**Pull request merged**